c#-ectoken/ecencryptstdlib/ECTokenGenerator.cs

```diff
@@ -0,0 +1,309 @@
+﻿using Org.BouncyCastle.Crypto;
+using Org.BouncyCastle.Crypto.Engines;
+using Org.BouncyCastle.Crypto.Modes;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Security;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace ecencryptstdlib
+{
+    /// <summary>
+    /// ECTokenGenerator for .NET Standard Library 1.4
+    /// ported and extended by Andreas Pollak (SpectoLogic)
+    /// </summary>
+    public class ECTokenGenerator
+    {
+        private static readonly SecureRandom Random = new SecureRandom();
+        private readonly Random _rand = new Random((int)DateTime.Now.Ticks);
+        private readonly int MIN_RANDOM_LENGTH = 4;
+        private readonly int MAX_RANDOM_LENGTH = 8;
+        private readonly char[] PADDING = { '=' };
+        //Preconfigured Encryption Parameters
+        public readonly int NonceBitSize = 96;
+        public readonly int MacBitSize = 128;
+        public readonly int KeyBitSize = 256;
+
+        public string NextRandomString(int length)
```

Comment:
@tinselcity If you are refering to ECTokenGenerator in general it is used by the c#-ectoken/ec_encryptcore/Program.cs Core Sample implementation.